packages/katello/katello/hostname-change.rb

```diff
@@ -239,6 +281,116 @@ def setup_opt_parser
       @opt_parser.parse!
     end
 
+    def dns_managed?
+      @scenario_answers['foreman_proxy']['dns'] &&
+        @scenario_answers['foreman_proxy']['dns_managed']
+    end
+
+    def should_update_dns?
+      dns_managed? && !@options[:skip_dns]
+    end
+
+    def query_dns_server(fqdn, zone, local_ip)
+      query_dns = Resolv::DNS.new(nameserver: [local_ip], search: [zone], ndots: 1)
+      a_record = query_dns.getresource(fqdn, Resolv::DNS::Resource::IN::A)
+      ip = a_record.address.to_s if a_record
+      soa_record = query_dns.getresource(zone, Resolv::DNS::Resource::IN::SOA)
+      serial = soa_record.serial if soa_record
+      {
+        ip: ip,
+        serial: serial,
+      }
+    end
+
+    def new_dns_values
+      STDOUT.puts 'gathering DNS data...'
+      new_vals = OpenStruct.new
+      new_vals.local_ip = '127.0.0.1'
+      new_vals.zone = @scenario_answers['foreman_proxy']['dns_zone']
+      new_vals.reverse_zone = @scenario_answers['foreman_proxy']['dns_reverse']
+      if new_vals.reverse_zone.is_a?(Array)
+        new_vals.reverse_zone = new_vals.reverse_zone.first
```

Comment:
I'm never happy with the pattern that you ignore the rest of an array. Should we instead loop over every one?

Reply:
Would there be more than one item in the array? And is it always an array? I'd like to remove the check for 'Array' if possible

Reply:
@jturel on the box I tested with it was a string, but for @chris1984 it was an array.  (or vice versa, can't remember)

Reply:
Updated to `[@scenario_answers['foreman_proxy']['dns_reverse']].flatten.first`